### PR TITLE
feat: support for watching API Gateway

### PR DESCRIPTION
**Pull request by RomainMuller**
Adds support for watching API Gateway `RestApi`'s.

### CODE REVIEW

File: lib/api-gateway.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import apigw = require('@aws-cdk/aws-apigateway');
+import { Metric, MetricOptions, ComparisonOperator, GraphWidget } from '@aws-cdk/aws-cloudwatch';
+import { Construct, Duration } from '@aws-cdk/core';
+import { IWatchful } from './api';
+
+export interface WatchApiGatewayRestApiOptions {
+  /**
+   * Alarm when 5XX errors reach this threshold over 5 minutes.
+   *
+   * @default - no alarm.
```
Comment (eladb) [inline lib/api-gateway.ts]: Why no alarm by default. I think we should have an alarm set up by default for 500s

File: lib/api-gateway.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        this.createCacheGraphWidget(operation),
+        this.createLatencyGraphWidget(ApiGatewayMetric.Latency, operation),
+        this.createLatencyGraphWidget(ApiGatewayMetric.IntegrationLatency, operation),
+      )
+    );
+  }
+
+  private createCallGraphWidget(opts?: WatchedOperation) {
+    return new GraphWidget({
+      title: `${opts ? `${opts.httpMethod} ${opts.resourcePath}` : 'Overall'} Calls`,
```
Comment (eladb) [inline lib/api-gateway.ts]: Add period in the title (see other resources, which I've just added)

File: lib/api-gateway.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      width: 12,
+      stacked: false,
+      left: [
+        this.createApiGatewayMetric(ApiGatewayMetric.Count, opts, { label: 'Calls', statistic: 'sum', color: '#1f77b4' }),
+        this.createApiGatewayMetric(ApiGatewayMetric.FourHundredError, opts, { label: 'HTTP 4XX', statistic: 'sum', color: '#ff7f0e' }),
+        this.createApiGatewayMetric(ApiGatewayMetric.FiveHundredError, opts, { label: 'HTTP 5XX', statistic: 'sum', color: '#d62728' }),
+      ],
+    });
```
Comment (eladb) [inline lib/api-gateway.ts]: Add 500s alarm annotation here if it's enabled

File: lib/api-gateway.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      );
+    }
+
+    this.watchful.addSection(props.title, {
+      links: [{ title: 'Amazon API Gateway Console', url: linkForApiGateway(props.restApi) }]
+    });
+    [undefined, ...props.watchedOperations || []].forEach(operation =>
+      this.watchful.addWidgets(
+        this.createCallGraphWidget(operation),
+        this.createCacheGraphWidget(operation),
```
Comment (eladb) [inline lib/api-gateway.ts]: I think the cache graph should be optional and disabled by default.

File: lib/watchful.ts [hunk 5]
```diff
@@ -67,6 +69,12 @@ export class Watchful extends Construct implements IWatchful {
     });
   }
 
+  public watchRestApi(title: string, restApi: apigw.RestApi, options: WatchApiGatewayRestApiOptions = {}) {
```
Comment (eladb) [inline lib/watchful.ts]: Rename to `watchApiGateway` and also the class to `WatchApiGateway`

File: lib/api-gateway.ts (hunk 1 above)
Comment (RomainMuller) [reply lib/api-gateway.ts]: Sure, making it default to `1`.

File: lib/api-gateway.ts (hunk 2 above)
Comment (RomainMuller) [reply lib/api-gateway.ts]: The pattern seems to make sense for sum aggregates (Counts / N minutes), however how'd you see this for non-sum aggregates (e.g: P90 latency)?

Comment (eladb) [reply lib/api-gateway.ts]: Not sure I understand, it's still accumulated over a certain period, isn't it?

**Pull request merged**